src/plone/restapi/services/users/add.py

```diff
@@ -86,3 +112,84 @@ def reply(self):
             ISerializeToJson
         )
         return serializer()
+
+    def _get_user(self, user_id):
+        portal = getSite()
+        portal_membership = getToolByName(portal, 'portal_membership')
+        return portal_membership.getMemberById(user_id)
+
+    def _error(self, status, type, message):
+        self.request.response.setStatus(status)
+        return {'error': {'type': type,
+                          'message': message}}
+
+    def update_password(self, data):
+        username = self.params[0]
+        target_user = self._get_user(username)
+        reset_token = data.get('reset_token', None)
+        old_password = data.get('old_password', None)
+        new_password = data.get('new_password', None)
+
+        pas = getToolByName(self.context, 'acl_users')
+        mt = getToolByName(self.context, 'portal_membership')
+        pwt = getToolByName(self.context, 'portal_password_reset')
+
+        if target_user is None:
+            self.request.response.setStatus(404)
+            return
+
+        # Send password reset mail
+        if data.keys() == []:
+            registration_tool = getToolByName(self.context,
+                                              'portal_registration')
+            registration_tool.mailPassword(username, self.request)
+            return
+
+        if reset_token and old_password:
+            return self._error(
+                400, 'Invalid parameters',
+                "You can't use 'reset_token' and 'old_password' together.")
+        if reset_token and not new_password:
+            return self._error(
+                400, 'Invalid parameters',
+                "If you pass 'reset_token' you have to pass 'new_password'")
+        if old_password and not new_password:
+            return self._error(
+                400, 'Invalid parameters',
+                "If you pass 'old_password' you have to pass 'new_password'")
+
+        # Reset the password with a reset token
+        if reset_token:
+            try:
+                pwt.resetPassword(username, reset_token, new_password)
+            except InvalidRequestError:
+                return self._error(403, 'Unknown Token',
+                                   'The reset_token is unknown/not valid.')
+            except ExpiredRequestError:
+                return self._error(403, 'Expired Token',
+                                   'The reset_token is expired.')
+            return
+
+        if old_password:
+            # Check permissions
+            # FIXME: Check manager permission
```

Comment:
@csenger what's the status here? can we check permissions properly? Otherwise we can not merge this pr into master anytime soon.